File: Services/UserManagement/UserManager/UserManager/Models/DTO/UserDto.cs
using Microsoft.Extensions.Diagnostics.HealthChecks;
using System.ComponentModel.DataAnnotations;

namespace UserManager.Models.DTO
{
    public class UserDto
    {
        public string? Id { get; set; } = string.Empty;

        [Required]
        public string UserName { get; set; }

        [Required]
        [EmailAddress]
        public string EmailAddress { get; set; }

        [Required]
        public string Password { get; set; }

        [Required]
        public string ConfirmPassword { get; set; }
    }
}

File: splitapp/src/app/Services/Auth/Models/UserDto.ts
export class UserDto{
    UserName: string;
    EmailAddress: string;
    Password: string;
    ConfirmPassword: string;

    constructor(UserName: string, EmailAddress: string, Password: string, ConfirmPassword: string){
        this.UserName = UserName;
        this.EmailAddress = EmailAddress;
        this.Password = Password;
        this.ConfirmPassword = ConfirmPassword;
    }
}